libpod/image/pull.go

```diff
@@ -356,11 +420,36 @@ func checkRemoteImageForLabel(ctx context.Context, label string, imageInfo pullR
 	if err != nil {
 		return err
 	}
+	return isLabelInInspect(label, remoteInspect.Labels, imageInfo.image)
+}
+
+// checkRemoteImageIsUpToDate checks if the remote image has a specific label. If not, an error is returned.
+// Then it checks if an source reference is newer than lastUpdated, or the time an image we are checking was last updated.
+// It returns false if lastUpdate is older than the source image, true otherwise
+func checkRemoteImageIsUpToDate(ctx context.Context, label *string, lastUpdated time.Time, imageInfo pullRefPair, sc *types.SystemContext) (bool, error) {
```

Comment:
I'd love to get this piped out to the image.Image struct so we can easily do a call like `image.IsLatest()` there and tell if the image is safe to use, but I'm fine with leaving that for another PR

Reply:
@mheon Only issue there is it would require two queries for the inspect in an image.New() (one for the labels, one to check the age). Hence both of those in the same function. I'd rather not have that run over the network twice.

Reply:
Actually, I guess the check latest part could be part of the image.IsLatest() function...

Reply:
Ultimately, this could be split this into two parts, one that does a `NewImage+Inspect`, and one that interprets the data.

(But until we have a _user_ for `image.IsLatest`, it seems fairly premature to worry about this.)